eli5/formatters/html.py

```diff
@@ -119,12 +120,14 @@ def render_targets_weighted_spans(targets, preserve_density):
 
 def render_weighted_spans(pws):
     # type: (PreparedWeightedSpans) -> str
-    # TODO - can be much smarter, join spans at least
-    # TODO - for longer documents, remove text without active features
+    # TODO - for longer documents, an remove text without active features
     return ''.join(
-        _colorize(token, weight, pws.weight_range)
-        for token, weight in zip(
-            pws.doc_weighted_spans.document, pws.char_weights))
+        _colorize(''.join(t for t, _ in tokens_weights),
+                  weight,
+                  pws.weight_range)
+        for weight, tokens_weights in groupby(
+            zip(pws.doc_weighted_spans.document, pws.char_weights),
```

Comment:
We're showing weights with .3f precision; what do you think about grouping spans if they have the same weight after formatting? It would obviously make it coupled with templates, but maybe it could help to reduce span count even more (I'm not sure how much though).